### ISSUE TITLE
micronaut  cookie max-age   session value

### ISSUE THREAD

**Issue by funnyleolee** [micronaut-projects/micronaut-security#339]
How set `micronaut.security.token.jwt.cookie.cookie-max-age` for session value
eg:
![github](https://user-images.githubusercontent.com/1265291/88899045-2b295f80-d280-11ea-838b-04bedf4508c5.png)

**Comment by jameskleeh** [comment]
This is currently not possible because if you don't set that value the expiration from the access token configuration will be used. That expiration setting has a default so it's not possible to not have a value. In order to use session no max age must be sent.

**Comment by jameskleeh** [comment]
@funnyleolee Can you describe why you would want to do that?

**Comment by funnyleolee** [comment]
> @funnyleolee Can you describe why you would want to do that?

I want token auto clean when browser closing. Now I use javascript add token to cookie, but not support httponly. If server side add that can be  httponly.however, I can just do my job on client side.

**Comment by Igor-Thaller** [comment]
Yes, we also thought about this situation. We want the cookie to expire with the session. 

Example: We have a Micronaut backend which is secured  with JWT provided in a cookie. The JWT expiration is 8 hours.
1) User A logs in on a shared computer (e.g. in school) and does some changes. Then user A closes the browser without a log off. 
2) User B opens the same page and can still use user A's JWT to perform some changes!

This situation would not arise if the cookie could be a session cookie (cookie expiration and cookie max age not set) because after closing the browser the cookie would be removed and user B would not have any chance to use user A's JWT.

**Comment by sdelamo** [comment]
@Igor-Thaller you could do a bean replacement and customize it: 

```java
public class TokenCookieLoginHandlerReplacement extends TokenCookieLoginHandler {
protected List<Cookie> getCookies(AccessRefreshToken accessRefreshToken, HttpRequest<?> request) {
//TODO
}
}

```
